staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2721,6 +2721,10 @@ const (
 	// TerminationByKubelet reason in DisruptionTarget pod condition indicates that the termination
 	// is initiated by kubelet
 	PodReasonTerminationByKubelet = "TerminationByKubelet"
+
+	// PodReasonPreemptionByKubeScheduler reason in DisruptionTarget pod condition indicates that the
+	// disruption was initiated by scheduler's preemption.
+	PodReasonPreemptionByKubeScheduler = "PreemptionByKubeScheduler"
```

Comment:
how are readers going to use this reason? if a custom scheduler pre-empted a pod, would we expect them to set this reason or a custom one, e.g. `PreemptionByMyCustomScheduler`?

Reply:
if this would be valid for other schedulers to set, then should be less specific, e.g. `PreemptionByScheduler`?

Reply:
> if a custom scheduler pre-empted a pod, would we expect them to set this reason or a custom one, e.g. PreemptionByMyCustomScheduler?

For consistency, it's preferred to just set the same reason. The question "by which scheduler" can be inferred by the pod's .spec.schedulerName.

> if this would be valid for other schedulers to set, then should this be less specific, e.g. PreemptionByScheduler?

LGTM. @mimowo do you see any side effects of changing it?

Reply:
I'm fine with the change

Reply:
Thanks @mimowo . I will raise PR to update website later: https://cs.k8s.io/?q=PreemptionByKubeScheduler&i=nope&files=&excludeFiles=vendor&repos=

Reply:
> For consistency, it's preferred to just set the same reason. The question "by which scheduler" can be inferred by the pod's .spec.schedulerName.

It seems we can't use `victims's .spec.schedulerName` to distinguish `by which scheduler`. It comes form the preemptor's spec.schedulerName.

Reply:
The message could be more flexible and include the name of the scheduler name.

Reply:
> It seems we can't use victims's .spec.schedulerName to distinguish by which scheduler. It comes form the preemptor's spec.schedulerName.

Yup, from the condition's message, we're able to know the preemptor pod:

https://github.com/kubernetes/kubernetes/blob/9b64025f36630af9dae1600e231e6b429ae6bda6/pkg/scheduler/framework/preemption/preemption.go#L365

Reply:
that... doesn't seem appropriate... putting the name of a pod maybe in another namespace into the condition of the preempted pod is a cross-namespace data leak, right?

Reply:
indicating kube-scheduler was the preemptor is good, but we should not be communicating pod names across namespaces like that

Reply:
> putting the name of a pod maybe in another namespace into the condition of the preempted pod is a cross-namespace data leak

Good point, especially in a multi-tenant cluster. But in non-multi-tenant clusters, it does provide valuable info to diagnose why a Pod is disrupted. So what's our take on this... (this message was introduced in 1.26)

Reply:
> So what's our take on this... 

Don't make kubernetes components write info across namespaces :)

Reply:
#114882